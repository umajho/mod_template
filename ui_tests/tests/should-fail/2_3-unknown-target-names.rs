fn main() {}

#[mod_template::define(macro_rules! define_foo; constructions(CONS -> ToCons), attribute_substitutions(ATTR_SUB))]
mod __ {
    #[__CONSTRUCT(foo as CONS)]
    fn good_construction() {}

    #[__SUBSTITUTE(ATTR_SUB)]
    fn good_attribute_substitution() {}

    #[__CONSTRUCT(foo as NOT_FOUND)]
    fn bad_construction_not_found() {}

    #[__SUBSTITUTE(NOT_FOUND)]
    fn bad_attribute_substitution_not_found() {}

    #[__CONSTRUCT(foo as ATTR_SUB)]
    fn bad_use_attribute_substitution_as_construction() {}

    #[__SUBSTITUTE(CONS)]
    fn bad_use_construction_as_attribute_substitution() {}
}
