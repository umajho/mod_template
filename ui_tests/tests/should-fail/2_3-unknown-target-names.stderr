error: unknown target name `NOT_FOUND`. It should be declared in the `constructions` block among the options of the attribute `mod_template::define`
  --> tests/should-fail/2_3-unknown-target-names.rs:11:26
   |
11 |     #[__CONSTRUCT(foo as NOT_FOUND)]
   |                          ^^^^^^^^^

error: unknown target name `NOT_FOUND`. It should be declared in the `attribute_substitutions` block among the options of the attribute `mod_template::define`
  --> tests/should-fail/2_3-unknown-target-names.rs:14:20
   |
14 |     #[__SUBSTITUTE(NOT_FOUND)]
   |                    ^^^^^^^^^

error: unknown target name `ATTR_SUB`. It should be declared in the `constructions` block among the options of the attribute `mod_template::define`
  --> tests/should-fail/2_3-unknown-target-names.rs:17:26
   |
17 |     #[__CONSTRUCT(foo as ATTR_SUB)]
   |                          ^^^^^^^^

error: unknown target name `CONS`. It should be declared in the `attribute_substitutions` block among the options of the attribute `mod_template::define`
  --> tests/should-fail/2_3-unknown-target-names.rs:20:20
   |
20 |     #[__SUBSTITUTE(CONS)]
   |                    ^^^^
